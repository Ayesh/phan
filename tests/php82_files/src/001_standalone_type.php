<?php
function test_true(true $x): true {
    '@phan-debug-var $x';
    return $x;
}
function test_nullabletrue(true|null $x): ?true {
    '@phan-debug-var $x';
    return $x;
}
function test_false(false $x): false {
    '@phan-debug-var $x';
    return $x;
}
function test_null(null $x): null {
    '@phan-debug-var $x';
    return $x;
}
var_export([
    test_true(true),
    test_true(null),
    test_nullabletrue(true),
    test_nullabletrue(false),
    test_nullabletrue(null),
    test_false(false),
    test_false(true),
    test_null(false),
    test_null(null),
]);
