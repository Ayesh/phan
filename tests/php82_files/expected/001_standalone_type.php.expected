%s:2 PhanDebugAnnotation @phan-debug-var requested for variable $x - it has union type true(real=true)
%s:6 PhanDebugAnnotation @phan-debug-var requested for variable $x - it has union type ?true(real=?true)
%s:10 PhanDebugAnnotation @phan-debug-var requested for variable $x - it has union type false(real=false)
%s:14 PhanDebugAnnotation @phan-debug-var requested for variable $x - it has union type null(real=null)
%s:14 PhanPluginUseReturnValueNoopVoid The function/method \test_null() is declared to return null and it has no side effects
%s:20 PhanTypeMismatchArgumentReal Argument 1 ($x) is null of type null but \test_true() takes true defined at %s:2
%s:22 PhanTypeMismatchArgument Argument 1 ($x) is false of type false but \test_nullabletrue() takes ?true defined at %s:6
%s:25 PhanTypeMismatchArgument Argument 1 ($x) is true of type true but \test_false() takes false defined at %s:10
%s:26 PhanTypeMismatchArgumentReal Argument 1 ($x) is false of type false but \test_null() takes null defined at %s:14
